tests/N3Tests/isImpliedBy/isImpliedBy_bcRule.nq

```diff
@@ -0,0 +1,6 @@
+?a <http://example.org/b1> <http://example.org/c1> _:b0 .
+?a <http://example.org/b> <http://example.org/c> _:b1 .
+_:b0 <http://www.w3.org/2000/10/swap/log#isImpliedBy> _:b1 _:b2 .
```

Comment:
> @jeswr yes I think that makes a lot of sense!

So it turns out that doesn't work on files like this because N3 doesn't fully support `n-quads` syntax, in particular having a graph term like this one.

Do you think it is worth opening up discussion as to whether this syntax should be supported in the spec @william-vw @josd .

Reply:
Yes @jeswr this makes a lot sense and the latest EYE does an experimental parsing of n-quads.
```
?a <http://example.org/b1> <http://example.org/c1> _:b0 .
?a <http://example.org/b> <http://example.org/c> _:b1 .
_:b0 <http://www.w3.org/2000/10/swap/log#isImpliedBy> _:b1 _:b2 .
```
is now `--pass`'ed as
```
<<_:sk_0 <http://example.org/b1> <http://example.org/c1>>> <http://www.w3.org/2000/10/swap/graph#name> _:e_b0_1.
<<_:sk_1 <http://example.org/b> <http://example.org/c>>> <http://www.w3.org/2000/10/swap/graph#name> _:e_b1_1.
<<_:e_b0_1 <= _:e_b1_1>> <http://www.w3.org/2000/10/swap/graph#name> _:e_b2_1.
```

Reply:
Generally the results of eval tests are given in an n-quads format (in particular see the [rdf-tests repo](https://github.com/w3c/rdf-tests/tree/main/rdf/rdf11)) as it is straightforward to parse and as such is a good syntax in which to unambiguously provide the baseline result.

In terms of what I'm proposing, it is that the following:

```n3
@prefix : <http://example.org/> .
"a" :b { :c :d :e } .
```

Be considered equivalent to:

```n3
@prefix : <http://example.org/> .
"a" :b _:b0 .
:c :d :e _:b0 .
```

Parsers over in the RDF/JS world like [N3.js](https://github.com/rdfjs/N3.js/) already interpret the first statement as the following RDF/JS quads:


```ts
const result = [
quad(
  literal("a"),
  namedNode("http://example.org/b"),
  blankNode("b0"),
),
quad(
  namedNode("http://example.org/c"),
  namedNode("http://example.org/d"),
  namedNode("http://example.org/e"),
  blankNode("b0"),
)]
```

Reply:
> In terms of what I'm proposing, it is that in N3 the following:
> 
> ```
> @prefix : <http://example.org/> .
> "a" :b { :c :d :e } .
> ```
> 
> Be considered equivalent to:
> 
> ```
> @prefix : <http://example.org/> .
> "a" :b _:b0 .
> :c :d :e _:b0 .
> ```

IMHO this is not the same!

Reply:
> In terms of what I'm proposing, it is that in N3 the following:
> 
> ```
> @prefix : <http://example.org/> .
> "a" :b { :c :d :e } .
> ```
> 
> Be considered equivalent to:
> 
> ```
> @prefix : <http://example.org/> .
> "a" :b _:b0 .
> :c :d :e _:b0 .
> ```

They are different, the former is a graph term, the latter is a graph by reference.
I have added initial support for n-quads in eye and this example round trips fine:
```
$ cat nq2.n3
@prefix : <http://example.org/> .

"a" :b _:b0 .
:c :d :e _:b0 .
```
```
$ eye --quiet --nope nq2.n3 --pass
@prefix : <http://example.org/>.

"a" :b _:e_b0_1.
:c :d :e _:e_b0_1.
```
One can of course compose graph term with the following rule in
```
$ cat nq3.n3
@prefix log: <http://www.w3.org/2000/10/swap/log#>.
@prefix graph: <http://www.w3.org/2000/10/swap/graph#>.
@prefix : <http://example.org/> .

"a" :b _:b0 .
:c :d :e _:b0 .
:f :g :h _:b0 .

{   ?A ?B ?C.
    ({?D ?E ?F} {?D ?E ?F ?C} ?L) log:collectAllIn ?SCOPE.
    ?G graph:list ?L.
} => {
    ?A ?B ?G.
}.
```
```
$ eye --quiet --nope nq3.n3 --pass
@prefix log: <http://www.w3.org/2000/10/swap/log#>.
@prefix graph: <http://www.w3.org/2000/10/swap/graph#>.
@prefix : <http://example.org/>.

"a" :b _:e_b0_1.
"a" :b {
    :c :d :e.
    :f :g :h.
}.
:c :d :e _:e_b0_1.
:f :g :h _:e_b0_1.
:c :d :e {
    :c :d :e.
    :f :g :h.
}.
:f :g :h {
    :c :d :e.
    :f :g :h.
}.
```

Reply:
> Generally the results of eval tests are given in an n-quads format (in particular see the [rdf-tests repo](https://github.com/w3c/rdf-tests/tree/main/rdf/rdf11)) as it is straightforward to parse and as such is a good syntax in which to unambiguously provide the expected result of parsing.

Why would n-quads be better to provide expected results - what would the graph label indicate? I'm not finding n-quads result files in the folder you reference (aside from the rdf-n-quads folder, but those are not result files). Not trying to be difficult here, I am just wondering what the added value would be for an individual results file. 

It's a much bigger discussion, for sure, but I feel that graph terms can be used to group triples belonging to the same graph (these could be made "referentially transparent" as well).